sw/airborne/subsystems/gps/gps_datalink.c

```diff
@@ -168,8 +192,14 @@ static void parse_gps_datalink(uint8_t numsv, int32_t ecef_x, int32_t ecef_y, in
   gps_datalink.last_3dfix_ticks = sys_time.nb_sec_rem;
   gps_datalink.last_3dfix_time = sys_time.nb_sec;
 
-  // publish new GPS data
   uint32_t now_ts = get_sys_time_usec();
+
+  // if selected, publish magnetometer data
+  if (GPS_DATALINK_USE_MAG) {
```

Comment:
can be replaced by `#if GPS_DATALINK_USE_MAG` since it doesn't change at runtime, same below